presto-docs/src/main/sphinx/develop/delete-and-update.rst

```diff
@@ -0,0 +1,203 @@
+  well as any other fields of the row that the connector will need to complete the ``DELETE`` operation.
+  For a JDBC connector, that rowId is usually the primary key for the table and no other fields are required.
+  For other connectors, the information needed to identify a row usually consists of multiple physical columns.
+  With the Hive connector and a Hive ACID table, for example, the ``DELETE`` rowId consists of the three ORC ACID columns
+  that identify the row.
+
+* ``beginDelete()``::
+
+    ConnectorTableHandle beginDelete(
+         ConnectorSession session,
+         ConnectorTableHandle tableHandle,
+         Map<String, ColumnHandle> columnHandles)
+
+  As the last step in creating the ``DELETE`` execution plan, the connector's ``beginDelete()`` method is called.
+  In addition to the ``session`` and ``tableHandle``, a map of all column hanedles is supplied.
+
+  ``beginDelete()`` performs any orchestration needed in the connector to start processing the ``DELETE``.
+  This orchestration varies from connector to connector.  In the Hive ACID connector, for example, ``beginDelete()``
+  checks that the table is transactional and starts a Hive Metastore transaction.
+
+  ``beginDelete()`` returns a ``ConnectorTableHandle`` with any added information the connector needs when the handle
+  is passed back to ``finishDelete()`` and the split generation machinery.  For most connectors, the returned table
+  handle contains a flag identifying the table handle as a table handle for a ``DELETE`` operation.
+
+* ``finishDelete()``::
+
+      void finishDelete(
+          ConnectorSession session,
+          ConnectorTableHandle tableHandle,
+          Collection<Slice> fragments)
+
+  During ``DELETE`` processing, the Presto engine accumulates the ``Slice`` collections returned by ``UpdatablePageSource.finish()``.
+  After all splits have been processed, the engine calls ``finishDelete()``, passing the table handle and that
+  collection of ``Slice`` fragments.  In response, the connector takes appropriate actions to complete the ``Delete`` operation.
+  Those actions might include committing the transaction, assuming the connector supports a transaction paradigm.
+
+``ConnectorMetadata`` ``UPDATE`` API
+====================================
+
+A connector implementing ``UPDATE`` must specify three ``ConnectorMetadata`` methods.
+
+* ``getUpdateRowIdColumnHandle``::
+
+   ColumnHandle getUpdateRowIdColumnHandle(
+        ConnectorSession session,
+        ConnectorTableHandle tableHandle,
+        List<ColumnHandle> updatedColumns)
+
+  The ``updatedColumns`` list contains column handles for all columns updated by the ``UPDATE`` operation in table column order.
+
+  The ColumnHandle returned by this method provides the "rowId" used by the connector to identify rows to be updated, as
+  well as any other fields of the row that the connector will need to complete the ``UPDATE`` operation.
```

Comment:
Should we overload the rowId columnHandle abstraction with the "any other fields of the row that the connector will need to complete the ``UPDATE`` operation" or have a separate abstraction for the same which could be optional? Ideally the RowIdColumnHandle should only identify the updated rowIds

Reply:
The challenge is that some connectors won't have any other values.  For example, the JDBC connector may not need any other column values besides the SET  expressions, while the Hive connector needs the values of all columns that are not the targets of SET expressions.

Reply:
You can think of the "row ID" field as being anything that the connector needs to pass between the table scan and the update/delete. I would agree that name "row ID" is a bit misleading with that definition, but from an API and implementation perspective, having a single column is simpler. (since with a `row` type, you can have any number of nested columns)

Reply:
makes sense from an implementation perspective, but probably we can rename it accordingly